frontend/src/metabase-lib/queries/utils/dataset.ts

```diff
@@ -31,27 +32,56 @@ export function normalizeFieldRef(fieldRef: FieldReference) {
 export function findColumnIndexForColumnSetting(
   columns: DatasetColumn[],
   columnSetting: TableColumnOrderSetting,
+  query?: Lib.Query,
 ) {
   const fieldRef = columnSetting.fieldRef;
   // NOTE: need to normalize field refs because they may be old style [fk->, 1, 2]
+  // TODO: use import { normalize } from "metabase-lib/queries/utils/normalize";
   const normalizedFieldRef = fieldRef ? normalizeFieldRef(fieldRef) : undefined;
   // first try to find by fieldRef
   if (normalizedFieldRef != null) {
-    const dimension = Dimension.parseMBQL(normalizedFieldRef);
-    const index = dimension
-      ? _.findIndex(columns, col =>
-          dimension.isSameBaseDimension(fieldRefForColumn(col)),
-        )
-      : -1;
+    let columnIndex: number;
 
-    if (index >= 0) {
-      return index;
+    if (!query) {
+      // TODO: remove once migration is completed
+      throw new Error("query is required to find column index");
```

Comment:
Is this entire block going to be removed or just the `throw`?
I'm trying to understand is there anything that prevents us from doing a complete migration.

Reply:
as we figured out during the pairing session, we need to double check with public questions where dataset_query is not available, if it provides correct results or not. but the plan is to remove `if(!query)` check completely